File: Codewars/CPP/findNextSquare.cpp
#include <math.h>
#include <iostream>

long int findNextSquare(long int sq) {
    long int rootOfNumero = sqrt(sq);
    if (rootOfNumero * rootOfNumero != sq)
    {
        return -1;
    }
    rootOfNumero++;
    return rootOfNumero * rootOfNumero;
}

int main () {
    std::cout <<  findNextSquare(144) << std::endl;
    std::cout <<  findNextSquare(169) << std::endl;
    std::cout <<  findNextSquare(89) << std::endl;
    return 0;
}

File: Codewars/Go/kata/findNextSquare.go
package kata

import "math"

func FindNextSquare(sq int64) int64 {
	var root int64 = int64(math.Sqrt(float64(sq)))
	if root * root != int64(sq) {
		return -1
	}

	return int64(math.Pow(float64(root + 1), float64(2)))
}